### ISSUE TITLE
Nested method definition inside `class << obj` is reported as offense

### ISSUE THREAD

**Issue by buehmann** [bbatsov/rubocop#4218]
One false positive of Lint/NestedMethodDefinition was not covered by the fix for #4185:
```ruby
class Foo
  def x(obj)
    class << obj
      def y
      end
    end
  end
end
```

```
bug.rb:4:7: W: Method definitions must not be nested. Use lambda instead.
      def y ...
      ^^^^^
```

## Expected behavior

I expect RuboCop not do register an offense for nested `def`s like these.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.48.0 (using Parser 2.4.0.0, running on ruby 2.3.1 x86_64-darwin15)
```
